### PR TITLE
rename sdsl-lite to SDSL (to use one single name)

### PR DESCRIPTION
**Pull request by mboisson**

### CODE REVIEW

**Comment by mboisson** [comment]
This PR has some commits which are shared with https://github.com/easybuilders/easybuild-easyconfigs/pull/11082

File: easybuild/easyconfigs/g/googletest/googletest-1.10.0-GCCcore-8.3.0.eb [hunk 1]
```diff
@@ -16,10 +16,14 @@ builddependencies = [
...
 ]
+# build twice, once for static, once for shared libraries
+configopts = ['', ' -DBUILD_SHARED_LIBS=ON ']
+separate_build_dir = True
 
 sanity_check_paths = {
-    'files': ['lib/libgtest.a', 'lib/libgmock.a'],
-    'dirs': ['include'],
+    'files': ['lib/libgmock.a', 'lib/libgmock_main.a', 'lib/libgtest.a', 'lib/libgtest_main.a',
+              'lib/libgmock.so', 'lib/libgmock_main.so', 'lib/libgtest.so', 'lib/libgtest_main.so'],
```
Comment (smoors) [inline easybuild/easyconfigs/g/googletest/googletest-1.10.0-GCCcore-8.3.0.eb:25]: @mboisson please replace `.so` with SHLIB_EXT (below also)?

File: easybuild/easyconfigs/s/SDSL/SDSL-2.1.1-20191211-GCCcore-8.3.0.eb [hunk 2]
```diff
@@ -26,10 +26,15 @@ sources = [
...
+patches = ['remove-googletest.patch', 'deactivate-failed-test.patch']
 checksums = [
     'ffab55af40d33a1a1705a693589d95ec31318270d58830a00d3723ee74b0c47b',  # SDSL-2.1.1-20191211.tar.gz
     '9dc9157a9a1551ec7a7e43daea9a694a0bb5fb8bec81235d8a1e6ef64c716dcb',  # googletest-1.10.0.tar.gz
     '211d4c64c13175a5020f4f16e418b9410facf5d22fb2cb7295a58887a61ae13f',  # libdivsufsort-2.0.1-20161201.tar.gz
+    '0ca26181a2d76ff48bc4ab83ac173ea5d033dbc315d4953ddd951fcbc7022290',  # remove-googletest.patch
+    '3d047b1d62244056537cd9dc8653bb8b17c918a96cf818780e24ca2052ead38e',  # deactivate-failed-test.patch
 ]
 
 builddependencies = [('CMake', '3.15.3')]
```
Comment (smoors) [inline easybuild/easyconfigs/s/SDSL/SDSL-2.1.1-20191211-GCCcore-8.3.0.eb:40]: @mboisson please add `binutils` builddep here?

File: easybuild/easyconfigs/s/SDSL/deactivate-failed-test.patch [hunk 3]
```diff
@@ -0,0 +1,12 @@
+diff -ru sdsl-lite-c32874cb2d8524119f25f3b501526fe692df29f4.orig/test/cst_byte_test.cpp sdsl-lite-c32874cb2d8524119f25f3b501526fe692df29f4/test/cst_byte_test.cpp
```
Comment (smoors) [inline easybuild/easyconfigs/s/SDSL/deactivate-failed-test.patch:1]: @mboisson please add author and short description to this patch?
also, please rename this file to SDSL-<version>_deactivate-failed-test.patch?

same for the other patch

**Comment by boegel** [comment]
@mboisson This needs a sync with `develop` now that #11082 is merged?

**Comment by mboisson** [comment]
I don't think it did, since commits were cherry-picked between the two. But I did it anyway.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 4 out of 4 (2 easyconfigs in this PR)
node368.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/90e6101771d7b8c3da28f287e161dd0f for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 4 out of 4 (2 easyconfigs in this PR)
node126.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/d9718fce5b425af56c8ab1edc5c8e90a for a full test report.

**Review by lexming — APPROVED** [review]
LGTM

**Comment by lexming** [comment]
Going in, thanks @mboisson !

**Pull request merged**